core/controller/src/main/scala/whisk/core/entitlement/Entitlement.scala

```diff
@@ -193,7 +199,14 @@ protected[core] abstract class EntitlementProvider(config: WhiskConfig, loadBala
         logging.error(this, s"failed while checking entitlement: ${t.getMessage}")
     } flatMap { isAuthorized =>
       if (isAuthorized) Future.successful({})
-      else Future.failed(RejectRequest(Forbidden))
+      else
+        Future.failed(
+          RejectRequest(
+            Forbidden,
+            Some(
+              ErrorResponse(
+                Messages.notAuthorizedtoAccessResource(resources.map(r => r.fqname).mkString(", ")),
+                transid))))
```

Comment:
I don't think this is particularly helpful - the priv check is over a set of resources, and `isAuthorized` is a predicate that must hold over all of them. At this point, you can't disambiguate which of the resources was the one that failed the check. So the message is a blanket failure accessing all resources instead of the possibly just one that failed.

Reply:
yep.  i wanted to only include the specific resource(s) in the set that were not accessible, but, as you mentioned, the lower level method performing the access check returned a single boolean as the all-or-nothing accessibility value.  if this is not a common use case, perhaps this message can be left as it currently is - without naming any particular resource; otherwise, additional changes will be needed to surface inaccessible resource(s).

Reply:
You could refactor the individual resource checks and resolve all the promises later where you can provide the extra information. I think that’s the sprit if the issue as it would apply to sequences.

Reply:
@rabbah - is there a way to use the CLI to cause an auth check on a set of resources?  i think i have a refactored approach, but i'm not sure that the current automated tests exercise this code path.
/cc @dubeejw

Reply:
I'd create a unit test where it's easier to mock this instead of trying to do it through the CLI.